src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -181,39 +168,29 @@ public PriorityScheduler(int corePoolSize, int maxPoolSize,
   public PriorityScheduler(int corePoolSize, int maxPoolSize,
                            long keepAliveTimeInMs, TaskPriority defaultPriority, 
                            long maxWaitForLowPriorityInMs, ThreadFactory threadFactory) {
-    ArgumentVerifier.assertGreaterThanZero(corePoolSize, "corePoolSize");
-    if (maxPoolSize < corePoolSize) {
-      throw new IllegalArgumentException("maxPoolSize must be >= corePoolSize");
-    }
-    
-    //calls to verify and set values
-    setKeepAliveTime(keepAliveTimeInMs);
-    setMaxWaitForLowPriority(maxWaitForLowPriorityInMs);
-    
+    this(new WorkerPool(threadFactory, corePoolSize, maxPoolSize, 
```

Comment:
You think I should add functions for the other storage objects?

I considered making them their own files, but I did not see any value to offering them to others.  The implementation is pretty specific for the PriorityScheduler (or any priority based thread pool I guess).  So I kept them as inner classes just because I did not want to pollute the package space with more classes.